client/src/legacy/ElementEditor/entwine.js

```diff
@@ -60,16 +62,35 @@ jQuery.entwine('ss', ($) => {
     },
 
     onunmatch() {
-      resetStores();
+      // Reset the store if the user navigates to a different part of the CMS
+      // or after submission if there are no validation errors
+      if (!$('.cms-edit-form').data('hasValidationErrors')) {
+        resetStores();
+      }
       ReactDOM.unmountComponentAtNode(this[0]);
     },
 
-    /**
-     * Invalidate cache after the form is submitted to force apollo to re-fetch.
-     */
     'from .cms-edit-form': {
-      onaftersubmitform() {
-        resetStores();
+      onaftersubmitform(event, data) {
+        const validationResultPjax = JSON.parse(data.xhr.responseText).ValidationResult;
+        if (validationResultPjax) {
+          const validationResult = JSON.parse(validationResultPjax.replace(/<\/?script[^>]*?>/g, ''));
+
+          // Reset redux store if form is succesfully submitted so apollo to refetches element data
+          // Do not reset if there are any validation errors because we want redux to hydrate the
+          // form, rather than then refetching which will return a value from the database.
+          // Instead the user should still see any modfied value they just entered.
+          if (validationResult.isValid) {
+            $('.cms-edit-form').data('hasValidationErrors', false);
+            resetStores();
+          } else {
+            $('.cms-edit-form').data('hasValidationErrors', true);
+          }
+        } else {
+          // No validation result returned
+          // assume everything is valid and reset stores so data is refetched.
+          resetStores();
+        }
```

Comment:
This is needed for compatibility with versions of silverstripe/admin that don't have the accompanying PR.

Reply:
I think I'd prefer to just tighten the constraint within elementals composer.json to `^1.13.2`

Reply:
Done. CI will break until admin is tagged.